### ISSUE TITLE
Add warning for mutable default argument

### ISSUE THREAD

**Issue by tweakimp** [PyCQA/pylint#3109]
Problem:
```python
def test(x=[], y=""):
    y += "a"
    x.append(y)
    print(x)


test()
test()
test()

>>> ['a']
>>> ['a', 'a']
>>> ['a', 'a', 'a']
```
Mutable default arguments can behave differently from what most people are expecting.
`x` is evaluated at function definition time, not when the function is called. This results in a list with one more element every call. Note that strings are not mutable, so this does not happen with `y`.

I think it would be helpful for people to be warned if they use a mutable type as default argument.

**Comment by PCManticore** [comment]
This is already handled by the `dangerous-default-value` check:

`a.py:1:0: W0102: Dangerous default value [] as argument (dangerous-default-value)`

**Comment by tweakimp** [comment]
Thank you for your answer, my editor blocked this warning on default :(
